src/main/java/seedu/address/model/person/Task.java

```diff
@@ -0,0 +1,29 @@
+package seedu.address.model.person;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+abstract class Task {
+    // Identity fields
+    private final Name name;
+    private final Phone time;
+    private final Address moduleCode;
+
+    Task(Name name, Phone time, Address moduleCode) {
```

Comment:
Perhaps change Address class to module code class after merging for module code field is done

Reply:
Sounds good!